tests/app/questionnaire/routing/test_when_rule_evaluator.py

```diff
@@ -0,0 +1,993 @@
+from datetime import datetime
```

Comment:
We need to think about what needs to be tested, as these tests are hard to digest. I don't think we need to test every operation with every combination of value source. A value source will resolve to a value. As long as we have tests to ensure the value source resolution works (covered by the value source resolver tests), we should only need to test that the rules work with value sources i.e. not all of the values that value sources can return.

Reply:
I have reduced the number of combinations that run and made all the test data in line to improve readability at the cost of duplication.

For date, I decided to keep the tests with multiple operators, as that is not tested in `test_operators` as a comparison only from the perspective that it resolves values.